src/snowflake/connector/compute_chunk_size.py

```diff
@@ -0,0 +1,69 @@
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+
+from __future__ import annotations
+
+import logging
+import math
+
+logger = logging.getLogger(__name__)
+
+# Amazon S3 multipart upload limits
+# https://docs.aws.amazon.com/AmazonS3/latest/userguide/qfacts.html
+CURRENT_CHUNK_SIZE = 8 * 1024 ** 2
+MAX_OBJECT_SIZE = 5 * 1024 ** 4
+MAX_PART_SIZE = 5 * 1024 ** 3
+MIN_PART_SIZE = 5 * 1024 ** 2
+MAX_PARTS = 10000
+
+
+class ChunkSizeCalculator:
```

Comment:
Why do we need a class instead of just a function?

Reply:
depending on usage of it, or even could just be some util functions

Reply:
Rewrote the class into a function